lib/tools/node-modules/node-modules-builder.ts

```diff
@@ -133,27 +133,165 @@ export class NodeModulesBuilder implements INodeModulesBuilder {
 			} else {
 				this.cleanNodeModules(absoluteOutputPath, platform);
 			}
 
 			const npmPluginPrepare = this.$injector.resolve(NpmPluginPrepare, {});
-			npmPluginPrepare.preparePlugins(resolvedDependencies, platform);
+			npmPluginPrepare.preparePlugins(productionDependencies, platform);
 		}).future<void>()();
 	}
 
+	public getProductionDependencies(projectPath: string) {
+		var deps: any = [];
+		var seen: any = {};
+
+		var pJson = path.join(projectPath, "package.json");
+		var nodeModulesDir = path.join(projectPath, "node_modules");
+
+		var content = require(pJson);
+
+		Object.keys(content.dependencies).forEach((key) => {
+			var depth = 0;
+			var directory = path.join(nodeModulesDir, key);
+
+			// find and traverse child with name `key`, parent's directory -> dep.directory
+			traverseChild(key, directory, depth);
+		});
+
+		return filterUniqueDirectories(deps);
+
+		function traverseChild(name: string, currentModulePath: string, depth: number) {
+			// check if key appears in a scoped module dependency
+			var isScoped = name.indexOf('@') === 0;
+
+			if (!isScoped) {
+				// Check if child has been extracted in the parent's node modules, AND THEN in `node_modules`
+				// Slower, but prevents copying wrong versions if multiple of the same module are installed
+				// Will also prevent copying project's devDependency's version if current module depends on another version
+				var modulePath = path.join(currentModulePath, "node_modules", name); // /node_modules/parent/node_modules/<package>
+				var exists = ensureModuleExists(modulePath);
+
+				if (exists) {
+					var dep = addDependency(deps, name, modulePath, depth + 1);
+
+					traverseModule(modulePath, depth + 1, dep);
+				} else {
+					modulePath = path.join(nodeModulesDir, name); // /node_modules/<package>
+					exists = ensureModuleExists(modulePath);
+
+					if(!exists) {
+						return;
+					}
+
+					var dep = addDependency(deps, name, modulePath, 0);
+
+					traverseModule(modulePath, 0, dep);
+				}
+
+			}
+			// module is scoped 
+			else {
+				var scopeSeparatorIndex = name.indexOf('/');
+				var scope = name.substring(0, scopeSeparatorIndex);
+				var moduleName = name.substring(scopeSeparatorIndex + 1, name.length);
+				var scopedModulePath = path.join(nodeModulesDir, scope, moduleName);
+
+				var exists = ensureModuleExists(scopedModulePath);
+
+				if (exists) {
+					var dep = addDependency(deps, name, scopedModulePath, 0);
+					traverseModule(scopedModulePath, depth, dep);
+				}
+				else {
+					scopedModulePath = path.join(currentModulePath, "node_modules", scope, moduleName);
+
+					exists = ensureModuleExists(scopedModulePath);
+
+					if (!exists) {
+						return;
+					}
+
+					var dep = addDependency(deps, name, scopedModulePath, depth + 1);
+					traverseModule(scopedModulePath, depth + 1, dep);
+				}
+			}
+
+			function traverseModule(modulePath: string, depth: number, currentDependency: any) {
```

Comment:
What is the difference when traversing a "child" and a "module". I think one of these functions needs a better name.